### PR TITLE
all tested

### LINKED ISSUE #30: sign in with firebase
##call sign in with firebase on submit sign in form

- [x] sign in to firebase account
- [x] show snackbar if firebase auth error
- wait for authstatechanged to redirect to lobby screen
- [x] test sign in with email & pword fn

- [ ] navigate between welcome screen, sign in, create account, all while having access to authStateStream
    - after navigator.push => go from sign in to create screen, how can still have access to auth stream??

### PR DESCRIPTION
**Pull request by tamari-gray**
fixes #30

### CODE REVIEW

**Review by nickmeinhold — COMMENTED** [review]
Awesome work mate!!!

Should "firebase_auth_test.dart" be "firebase_auth_service_test.dart"?

**Comment by tamari-gray** [comment]
yeah good find!

**Comment by tamari-gray** [comment]
dont review yet, Im gonna test navigating to lobby page after sign in + wait for auth state change

**Comment by tamari-gray** [comment]
Hi @nickmeinhold, can i schedule a call tommorow night or between 3-5pm? im having a real hard time routing from sign in to lobby. It should be an easy task but ive been going in cirlces for 2 nights.

**Comment by nickmeinhold** [comment]
@tamari-gray any time mate! I usually go for a walk somewhere around 4 or 5 but I can go any time so if you want to pick a time I can work around it, and otherwise I'll be available all day at this stage. I hope I haven't stuffed you up with setting up the NavigationService the way we did. We'll get it sorted anyway! Cheers mate

**Comment by tamari-gray** [comment]
ready for review please @nickmeinhold

**Comment by nickmeinhold** [comment]
On it!

File: test/auth_section/sign_in/widget_test.dart
Comment (nickmeinhold) [inline test/auth_section/sign_in/widget_test.dart]: Personally I think your mocks should use variable names that include 'mock', eg. I reckon `authService` here
 should be `mockAuthService`. It's especially important I would say where you are using both real services and mock services.

**Review by nickmeinhold — DISMISSED** [review]
Looks awesome mate, great job!!!

**Comment by nickmeinhold** [comment]
@tamari-gray not sure if you want to change the mock variable names before merging so I won't merge and let you decide. Cheers mate!

File: test/auth_section/sign_in/widget_test.dart
Comment (tamari-gray) [reply test/auth_section/sign_in/widget_test.dart]: Yeah good point! Thank you :) It would be good to differentiate those as much as possible.

**Comment by tamari-gray** [comment]
hi @nickmeinhold, just gave the test file a good clean. And realised what happens if we have navigated between create account and sign in, then after a successfull login we popback to create account.
im thinking of doing to popuntil index:1 (initial route) because its easier, but it feels alot better to just navigate between create account & sign in via some state, then when we pop, wont have to see animation of popping multiple screens.
ok will just do the latter in a seperate ticket lol. 
thanks for the review anyway!

**Comment by tamari-gray** [comment]
oops need another review plz @nickmeinhold. thank you :pray:

**Review by nickmeinhold — APPROVED** [review]
Looks great man!  Digging the clean up 👍

**Pull request merged**
